src/Xamarin.Android.Build.Tasks/Tasks/GetAotArguments.cs

```diff
@@ -228,120 +226,23 @@ protected void GetAotOptions (NdkTools ndk, AndroidTargetArch arch, int level, s
 			if (SequencePointsMode == SequencePointsMode.Offline)
 				MsymPath = outdir;
 
-			string ldName;
-			if (EnableLLVM) {
-				ldName = ndk.GetToolPath (NdkToolKind.Linker, arch, level);
-				if (!string.IsNullOrEmpty (ldName)) {
-					ldName = Path.GetFileName (ldName);
-					if (ldName.IndexOf ('-') >= 0) {
-						ldName = ldName.Substring (ldName.LastIndexOf ("-", StringComparison.Ordinal) + 1);
-					}
-				}
-			} else {
-				ldName = "ld";
-			}
+			LdName = "ld";
+
 			string ldFlags = GetLdFlags (ndk, arch, level, toolPrefix);
-			if (!string.IsNullOrEmpty (ldName)) {
-				LdName = ldName;
-			}
 			if (!string.IsNullOrEmpty (ldFlags)) {
 				LdFlags = ldFlags;
 			}
 		}
 
-		string GetLdFlags(NdkTools ndk, AndroidTargetArch arch, int level, string toolPrefix)
+		string GetLdFlags (NdkTools ndk, AndroidTargetArch arch, int level, string toolPrefix)
 		{
-			var toolchainPath = toolPrefix.Substring (0, toolPrefix.LastIndexOf (Path.DirectorySeparatorChar));
-			var ldFlags = string.Empty;
-			if (EnableLLVM) {
-				string androidLibPath = string.Empty;
-				try {
-					androidLibPath = ndk.GetDirectoryPath (NdkToolchainDir.PlatformLib, arch, level);
-				} catch (InvalidOperationException ex) {
-					Diagnostic.Error (5101, ex.Message);
-				}
-
-				string toolchainLibDir;
-				if (ndk.UsesClang) {
-					if (ndk.NoBinutils) {
-						toolchainLibDir = String.Empty;
-					} else {
-						toolchainLibDir = GetNdkToolchainLibraryDir (ndk, toolchainPath, arch);
-					}
-				} else
-					toolchainLibDir = GetNdkToolchainLibraryDir (ndk, toolchainPath);
-
-				var libs = new List<string> ();
-				if (ndk.UsesClang) {
-					if (!String.IsNullOrEmpty (toolchainLibDir)) {
-						libs.Add ($"-L{toolchainLibDir.TrimEnd ('\\')}");
-					}
-					libs.Add ($"-L{androidLibPath.TrimEnd ('\\')}");
-
-					if (arch == AndroidTargetArch.Arm) {
-						// Needed for -lunwind to work
-						string compilerLibDir = Path.Combine (toolchainPath, "..", "sysroot", "usr", "lib", ndk.GetArchDirName (arch));
-						libs.Add ($"-L{compilerLibDir.TrimEnd ('\\')}");
-					}
-				}
-
-				if (!String.IsNullOrEmpty (toolchainLibDir)) {
-					libs.Add (Path.Combine (toolchainLibDir, "libgcc.a"));
-				}
-				libs.Add (Path.Combine (androidLibPath, "libc.so"));
-				libs.Add (Path.Combine (androidLibPath, "libm.so"));
-
-				ldFlags = $"\\\"{string.Join ("\\\";\\\"", libs)}\\\"";
-			}
-
-			if (!StripLibraries) {
-				return ldFlags;
-			}
-
```

Comment:
@grendello what are your thoughts on this? Do you remember why we needed to pass these extra lib paths from the NDK, and do you think they would still be needed if we use the `ld` that we now bundle and ship?

I tested template project with AOT+LLVM on a few devices and emulators (including an armeabi Nexux 5 running Android 6.0), and I haven't hit any issues yet...

Reply:
As far as I recall, the extra lib paths were for NDK versions which used `gcc`, as we needed to link private gcc libraries.  For the LLVM-based NDK we probably no longer need it, and our native toolchain doesn't need it too.  The only reason to keep them, I think, is to keep support for old NDK versions installed on user machines (we had an instance when somebody still used NDK r16). Other than that, I **think** they can be safely removed.

Reply:
Should we even continue to allow usage of an arbitrary Android NDK for our AOT tasks if our installers now contain all of the required tools?

Reply:
We have no influence on what version of NDK is installed on the user's machine, so we should support as many of them as possible

Reply:
I guess that's my question, is there a strong reason to allow using an external NDK instead of our toolchain at this point in time?

Reply:
AOT tasks should work just fine with our toolchain, it would be good to make that non-optional. Otherwise, we need to keep allowing NDK because of projects which include native code that requires the full C/C++ toolchain.

Reply:
Rationale for why we may need to support older NDK tooling, e.g. https://github.com/xamarin/xamarin-android/issues/6454

Reply:
By default, yes, we should use our embedded toolchain, ignoring `$(AndroidNdkDirectory)`. But we must make it possible for people to use their NDK if they need to because of compiler bugs, toolchain version requirements (for instance for reproducible builds). I can't find another issue we had when someone had to use NDK r16b because newer versions broke some code for them. For these reasons, even though I'd personally prefer to rely only on our embedded toolchain, we need to keep external NDK support in some form.